lib/schema/validator2.ex

```diff
@@ -307,22 +307,59 @@ defmodule Schema.Validator2 do
       if is_binary(version) do
         schema_version = Schema.version()
 
-        if version != schema_version do
-          add_error(
-            response,
-            "version_incorrect",
-            "Incorrect version at \"metadata.version\"; value of \"#{version}\"" <>
-              " does not match schema version \"#{schema_version}\"." <>
-              " This can result in incorrect validation messages.",
-            %{
-              attribute_path: "metadata.version",
-              attribute: "version",
-              value: version,
-              expected_value: schema_version
-            }
-          )
-        else
-          response
+        # Check if version matches OCSF's versioning format (x.y.z, x.y.z-dev, x.y.z-rc.n)
+        # Allows: 1.0.0, 1.5.0-dev, 1.0.0-rc.2 etc.
+        case String.match?(version, ~r/^\d+\.\d+\.\d+(?:-dev|-rc\.\d+)?$/) do
```

Comment:
This is minor, but I thought I'd ask:

Technically speaking, the [docs](https://github.com/ocsf/ocsf-docs/blob/fd5313de86114ec352f95ffab0998280188d5acd/FAQs/Schema%20FAQ.md?plain=1#L140) say OCSF follows SemVer.

`-dev` and `-rc.2` are patterns we follow today, but in theory there's nothing stopping us from using a different suffix in the future.  Are we committed to this limited scheme or is there a possibility we'll expand to other SemVer-valid strings?  And should this regex be one of the [suggested regexes to check SemVer strings](https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string)?  Imagining a situation down the road where if we forget this constraint was coded in, would we run the risk of accidentally breaking version validation?